### PR TITLE
VCS: remove unused methods and make new Git pattern the default

### LINKED ISSUE #8940: Build: cleanup VCS support to remove unused methods
Since the deploy that implements the Celery Handlers, we are always starting the build from a fresh and cleaned environment. No cache is used at all.

So, we should be able to delete all the code related to _update the repository_ by calling `git fetch` and similar. Take a look at https://github.com/readthedocs/readthedocs.org/blob/b5a05f2a474389620a1c51eeb139316aecb2757a/readthedocs/vcs_support/backends/git.py#L154-L176 as example.

### PR DESCRIPTION
**Pull request by humitos**
This PR removes the feature flag to use the new Git pattern, making it the default. Besides, it removes some methods that are not used anymore because we are always cloning the repository, so there is no need to update it or similar.

Closes #8940 

Related
*  #9108 
* #9367 
* #8840 
* https://github.com/readthedocs/readthedocs.org/pull/10594

### CODE REVIEW

**Comment by humitos** [comment]
There were some changes in our code, but this PR is still useful. I'll try to come back to it when I have some extra time.

**Comment by humitos** [comment]
I updated this PR but there are still some extra work to do, so I'm leaving it again 😄 

IMO, it would be good to deprecate and remove some VCS as already proposed (mainly SVN and bzr) to reduce the complexity and scope of this refactor.

Also, we should probably avoid keep using `gitpython` and go back to parse git's output (I think @stsewd already mentioned this in another issue). However, this should be simpler now because we are already doing this with `ls-remote` and it's working fine. Branches _and_ tags will use the same command to get these. The only new command we will be introducing to parse its output would be `git submodule` to discover them.

**Comment by humitos** [comment]
I will update this PR once Santos merges the work from https://github.com/readthedocs/readthedocs.org/pull/10594 so we don't compete for this chunk of code.

**Comment by humitos** [comment]
Blocked by https://github.com/readthedocs/readthedocs.org/pull/10606

File: readthedocs/vcs_support/backends/git.py
Comment (stsewd) [inline readthedocs/vcs_support/backends/git.py:172]: I don't think we are using this currently, this value is never changed

**Review by stsewd — APPROVED** [review]
We should merge this after we have tested the new implementation more broadly again.

**Comment by emmettbutler** [comment]
Is it possible to tell a build to use the default prior to this change? It seems possible that this change broke dd-trace-py's release notes workflow. https://github.com/DataDog/dd-trace-py/issues/7176

**Pull request merged**
